test/e2e/qat/qatplugin_dpdk.go

```diff
@@ -165,6 +168,59 @@ func describeQatDpdkPlugin() {
 			ginkgo.It("does nothing", func() {})
 		})
 	})
+
+	ginkgo.Context("When a QAT device goes unresponsive", func() {
+		ginkgo.When("QAT's auto-reset is off", func() {
+			ginkgo.BeforeEach(func() {
+				ginkgo.By("creating a configMap before plugin gets deployed")
+				e2ekubectl.RunKubectlOrDie(f.Namespace.Name, "create", "configmap", "--from-literal", "qat.conf=$'ServiceEnabled=dc\nAutoresetEnabled=off", "qat-config")
+
+				ginkgo.By("setting resourceName for dc services")
+				resourceName = dcResource
+			})
+
+			ginkgo.It("checks if the heartbeat is read correctly [Functionality:heartbeat]", func(ctx context.Context) {
+				injectError(ctx, f, resourceName)
+
+				ginkgo.By("seeing if there is zero resource")
+				if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, resourceName, 100*time.Second, utils.WaitForZeroResource); err != nil {
+					framework.Failf("unable to wait for nodes to have no resource: %v", err)
+				}
+			})
+		})
+
+		ginkgo.When("QAT's autoreset is on", func() {
+			ginkgo.BeforeEach(func() {
+				ginkgo.By("creating a configMap before plugin gets deployed")
+				e2ekubectl.RunKubectlOrDie(f.Namespace.Name, "create", "configmap", "--from-literal", "qat.conf=$'ServiceEnabled=dc\nAutoresetEnabled=on", "qat-config")
+
+				ginkgo.By("setting resourceName for dc services")
+				resourceName = dcResource
+			})
+
+			ginkgo.It("checks if an injected error gets solved [Functionality:auto-reset]", func(ctx context.Context) {
+				injectError(ctx, f, resourceName)
+
+				ginkgo.By("seeing if there is zero resource")
+				if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, resourceName, 100*time.Second, utils.WaitForZeroResource); err != nil {
+					framework.Logf("unable to wait for nodes to have no resource: %v", err)
+					// It should not be Failf since there can be a case auto-reset worked before plugin notices.
+					// It is still necessary to have utils.WaitForNodeWithResource for WaitForZeroResource
+					// because there can be also a case that the following part that checks WaitForPositiveResource may run
+					// before the injected error is recognized by plugin when this part does not exist.
+					// In other words, this is necessary to ensure that the injected error does not remain after this It() spec.
```

Comment:
what if there is an actual error in the wait loop?

As far as I understand this, it still sounds very fragile/hacky to me. the test is about "injected error gets solved" but you are not interested in making sure that you can actually see the error first ("It should not be Failf since there can be a case auto-reset worked before plugin notices."). However, you are still adding some strange logic to "ensure that the injected error does not remain after this It() spec.".

Reply:
Actually, I realized that the logic is even not necessary anymore. We can now ensure noResource since we check positiveResource in `JustBeforeEach`. Let me remove the comment and change to Failf.